chainstate/src/detail/tests/normal_tokens.rs

```diff
@@ -0,0 +1,419 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author(s): A. Sinitsyn
+
+use super::{anyonecanspend_address, setup_chainstate};
+use crate::{
+    detail::{CheckBlockError, CheckBlockTransactionsError},
+    BlockError, BlockSource, Chainstate,
+};
+use chainstate_types::block_index::BlockIndex;
+use common::{
+    chain::{
+        block::{timestamp::BlockTimestamp, Block, ConsensusData},
+        signature::inputsig::InputWitness,
+        tokens::{token_id, AssetData, OutputValue, TokenId},
+        OutputPurpose, Transaction, TxInput, TxOutput,
+    },
+    primitives::{Amount, Idable},
+};
+use std::vec;
+
+macro_rules! assert_token {
+    ($block_index:expr, $err_name:ident) => {
+        assert!(matches!(
+            $block_index,
+            Err(BlockError::CheckBlockFailed(
+                CheckBlockError::CheckTransactionFailed(CheckBlockTransactionsError::$err_name(
+                    _,
+                    _
+                ))
+            ))
+        ));
+    };
+}
```

Comment:
Am I right that you are using macro here to preserve the exact line of the (potential) failure? In that case a function with the `#[track_caller]` attribute can be used, too. Personally, I prefer to fallback to a macro only when the same thing cannot be expressed without a boilerplate by other language tools. I'm not sure about this case, though.

Reply:
Removed assert_token